nuxeo-core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/TestSQLRepositoryAPI.java

```diff
@@ -1227,6 +1227,31 @@ public void testGetParentDocument() {
         assertEquals(root.getPathAsString(), shouldBeRoot.getPathAsString());
     }
 
+    @Test
+    public void testGetParentDocumentFromPathReference() {
+        DocumentModel root = session.getRootDocument();
+        DocumentModel shouldBeNull = session.getParentDocument(new PathRef(root.getPathAsString()));
+        assertNull(shouldBeNull);
+
+        String name = "folder#" + generateUnique();
+        DocumentModel parentFolder = session.createDocumentModel(root.getPathAsString(), name, "Folder");
+        DocumentModel shouldBeRoot = session.getParentDocument(parentFolder.getRef());
+        assertEquals(root.getPathAsString(), shouldBeRoot.getPathAsString());
+
+        parentFolder = session.createDocument(parentFolder);
+        session.save();
+        String name2 = "file#" + generateUnique();
+        DocumentModel childFile = session.createDocumentModel(parentFolder.getPathAsString(), name2, "File");
+        DocumentModel shouldBeFolder = session.getParentDocument(childFile.getRef());
+        assertEquals(parentFolder.getPathAsString(), shouldBeFolder.getPathAsString());
+
+        DocumentRef shouldBeRootRef = session.getParentDocumentRef(parentFolder.getRef());
```

Comment:
Maybe you could also test `getParentDocumentRef` on the root document (`new PathRef("/")`), as you are testing `getParentDocument` on it.